modules/gapi/src/backends/cpu/gcpubackend.cpp

```diff
@@ -224,5 +285,10 @@ void cv::gimpl::GCPUExecutable::run(std::vector<InObj>  &&input_objs,
         }
     } // for(m_script)
 
+    if (m_needKernelsSetup)
+    {
+        m_needKernelsSetup = false;
+    }
```

Comment:
Again, all this logic about handling the reset condition can be moved to a separate function. You don't need to spread these details everywhere.

Reply:
I moved this logic upper in the "run" code. Is this acceptable solution or better to move out logic into a separate function?